src/gui/KMessageWidget.cpp

```diff
@@ -287,7 +287,7 @@ void KMessageWidget::setMessageType(KMessageWidget::MessageType type)
             bg1 = palette().highlight().color();
             break;
         case Warning:
-            bg1.setRgb(181, 102, 0); // values taken from kcolorscheme.cpp (Neutral)
+            bg1.setRgb(191, 126, 7); // values taken from kcolorscheme.cpp (Neutral)
```

Comment:
I recommend the following code instead (lighter bg, dark fg) for warnings. I am also really not a fan of the gradient, its a little aggressive and out of place with the rest of the UI.

```
    QColor bg0, bg1, bg2, border;
    QColor fg = palette().light().color();
    switch (type) {
        case Positive:
            bg1.setRgb(0, 110,  40); // values taken from kcolorscheme.cpp (Positive)
            break;
        case Information:
            bg1 = palette().highlight().color();
            break;
        case Warning:
            bg1.setRgb(214, 190, 8);
            fg = Qt::black;
            break;
        case Error:
            bg1.setRgb(191, 3, 3); // values taken from kcolorscheme.cpp (Negative)
            break;
    }
    
    // Colors
    bg0 = bg1.lighter(110);
    bg2 = bg1.darker(110);
    border = darkShade(bg1);
```

Reply:
I think this whole widget should get a little redesign for 2.4.0. It served us well, but it needs a refresh.